src/platform/datapath_raw_xdp.c

```diff
@@ -957,6 +1057,61 @@ CxPlatDpRawUninitialize(
     }
 }
 
+_IRQL_requires_max_(PASSIVE_LEVEL)
+void
+CxPlatDpRawOnSocketStateChange(
+    _In_ CXPLAT_SOCKET* Socket,
+    _In_ BOOLEAN IsCreated
+    )
+{
+    XDP_DATAPATH* Xdp = (XDP_DATAPATH*)Socket->Datapath;
+
+    if (Socket->Wildcard) {
+        const XDP_RULE Rule = {
+            .Match = XDP_MATCH_UDP_DST,
+            .Pattern.Port = Socket->LocalAddress.Ipv4.sin_port, // TODO - Check on on byte order
```

Comment:
Looks like XDP compares this to the field in the actual packet header, i.e. expects the port in network order, so (assuming this is a normal sockaddr with port in network order) I think this is correct.